src/util.cc

```diff
@@ -423,21 +426,132 @@ const char* SpellcheckString(const char* text, ...) {
 }
 
 #ifdef _WIN32
+std::wstring Utf8ToWide(const StringPiece& u8_string) {
+  if (u8_string.len_ == 0)
+    return std::wstring();
+
+  // Make sure that the string size can be casted to an int.
+  assert(u8_string.size() <= (std::numeric_limits<int>::max)());
+
+  size_t size_needed = MultiByteToWideChar(
+      CP_UTF8, 0, u8_string.str_, static_cast<int>(u8_string.size()), NULL, 0);
+  if (size_needed == 0)
+    Fatal("Failed to get needed size of wide string");
+
+  std::wstring w_string(size_needed, 0);
+  int w_size_needed = MultiByteToWideChar(CP_UTF8, 0, u8_string.str_,
+                                          static_cast<int>(u8_string.size()),
+                                          &w_string[0], w_string.size());
+  if (w_size_needed != size_needed)
+    Fatal("Failed conversion to wide string");
+
+  return w_string;
+}
+
+int GetUtf8SizeNeeded(const WStringPiece& w_string) {
+  // Make sure that the string size can be casted to an int.
+  assert(w_string.size() <= (std::numeric_limits<int>::max)());
+
+  int size_needed = WideCharToMultiByte(CP_UTF8, 0, w_string.wstr_,
+                                        static_cast<int>(w_string.size()), NULL,
+                                        0, NULL, NULL);
+  if (size_needed == 0)
+    Fatal("Failed to get needed size of UTF-8 string");
+
+  return size_needed;
+}
+
+std::string WideToUtf8(const WStringPiece& w_string) {
+  if (w_string.len_ == 0)
+    return string();
+
+  // Make sure that the string size can be casted to an int.
+  assert(w_string.size() <= (std::numeric_limits<int>::max)());
+
+  int size_needed = GetUtf8SizeNeeded(w_string);
+
+  std::string u8_string(size_needed, '\0');
+  int u8_size_needed = WideCharToMultiByte(
+      CP_UTF8, 0, w_string.wstr_, static_cast<int>(w_string.size()),
+      &u8_string[0], size_needed, NULL, NULL);
+  if (u8_size_needed != size_needed)
+    Fatal("Failed conversion to UTF-8 string");
+
+  return u8_string;
+}
+
+void ConvertWideToUtf8(const WStringPiece& w_string, int size_needed,
+                       char* buff, size_t buffer_length) {
+  if (size_needed > (buffer_length - 1)) // Also account for \0
```

Comment:
I don't see why you're passing in "size_needed".

You're only using it to compare the buffer length.

Instead, you should use the buffer length directly for these things.

Make it the responsibility of the calling function (e.g., clearly document this requirement), to provide a buffer that's *at least* large enough for the data.